### DISCUSSION TITLE
Support for AMD Ryzen AI NPU (on Linux)

### DISCUSSION THREAD

**Discussion by siccovansas** [nextcloud/app_api#445] [Q&A]
AMD releases more and more processors with [Ryzen AI NPUs](https://www.amd.com/en/products/processors/consumer/ryzen-ai.html). AMD has been slow with creating the required XDNA Linux driver, but they are working on it and it is [expected to be available soon](https://www.phoronix.com/news/AMD-XDNA-Linux-Driver-v4).

I'm new to AppAPI, but it seems that the current possible compute devices are CPU, CUDA or ROCm. What is needed for AppAPI to support Ryzen AI NPU/XDNA? It would be great if these Ryzen processors can use their NPUs to speed up Nextcloud AI tasks!

Another question. A Ryzen 8600G has an NPU and the 760M iGPU. There is currently [no ROCm support for the 760M iGPU](https://github.com/ROCm/ROCm/issues/3398), but if/once this is added would AppAPI be able to use both the NPU and iGPU (and even CPU) at the same time and maybe even for the same task?
UPDATE: I just [read in the FAQ](https://docs.nextcloud.com/server/latest/admin_manual/exapps_management/AppAPIAndExternalApps.html#faq) that you can select different compute devices for different apps (but not multiple devices for the same app). AMD is also working on a [Unigie AI Software Stack](https://www.phoronix.com/news/AMD-Unified-AI-Software-Stack), which might make it possible to automatically chose the compute device based on the task.

**Comment by oleksandr-nc** [comment]
For this we first need support from `pytorch`. Let's see how the situation develops, of course we are interested in supporting something other than Nvidia.
